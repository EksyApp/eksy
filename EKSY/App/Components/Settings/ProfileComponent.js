import React, {Component} from 'react'
import Label from "../Common/Label";
import Button from "../Common/Button";
import {View} from "react-native";
import Message from "../Common/Message";
import {Actions} from 'react-native-router-flux'

export default class ProfileComponent extends Component {
	
	renderUserInfo() {
		if (this.props.user) {
			return (
					<Message>{this.props.user.firebaseUser.displayName}</Message>
			)
		}
	}
	
	renderProfileActionButton() {
		if (this.props.user) {
			return (
					<Button onPress={() => {this.props.onSignoutClick()}}>
						Signout
					</Button>
			)
		} else {
			return(
					<Button onPress={() => {Actions.login()}}>Login</Button>
			)
		}
	}
	
	renderShowMarkersButton() {
		if (this.props.user) {
			return (
					<Button onPress={() => {this.props.onMarkersClick()}}>
						Markers
					</Button>
			)
		}
	}
	
	render() {
		return (
				<View>
					<Label>Profile</Label>
					{this.renderUserInfo()}
					{this.renderShowMarkersButton()}
					{this.renderProfileActionButton()}
				</View>
		)
	}
	
	
}