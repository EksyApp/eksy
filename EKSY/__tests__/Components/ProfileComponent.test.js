import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ProfileComponent from "../../App/Components/Settings/ProfileComponent";

test('ProfileComponent renders correctly', () => {
	const tree = renderer.create(
			<ProfileComponent />
	).toJSON();
	expect(tree).toMatchSnapshot();
});