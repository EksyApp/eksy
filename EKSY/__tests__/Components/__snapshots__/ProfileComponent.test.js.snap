// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileComponent renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgb(101,114,122)",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Profile
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "rgb(230,232,227)",
        "borderColor": "rgb(101,114,122)",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 15,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "center",
          "color": "rgb(101,114,122)",
          "fontSize": 16,
          "fontWeight": "600",
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      Login
    </Text>
  </View>
</View>
`;
